### PR TITLE
[WINDUP-3313] Allow for <project> dependencies to be detected depending on their location

### PR DESCRIPTION
**Pull request by jmle**
See [WINDUP-3313](https://issues.redhat.com/browse/WINDUP-3313).

### CODE REVIEW

**Comment by carlosthe19916** [comment]
The PR looks good to me, once the tests failures are fixed I will do a second round of review.

**Comment by jmle** [comment]
@carlosthe19916 this is finally ready for review - for some reason the build had failed momentarily.

**Review by carlosthe19916 — COMMENTED** [review]
@jmle Thanks for the PR. I've verified that the PR successfully identifies dependencies in:
- `dependencies` tag
- `plugins` tag
- `dependencyManagement` tag

The only place I saw it is not working is for dependencies located in a `profile` tag. Eg:

```

<profiles>
        <profile>
            <id>native</id>           
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.apache.maven.plugins</groupId>
                        <artifactId>maven-failsafe-plugin</artifactId>
                        <version>${surefire-plugin.version}</version>                        
                    </plugin>
                </plugins>
            </build>
        </profile>
```

Giving the fact that the `profiles` tag was not specified in the Ticket we can leave it for another PR. 

There is a very minor comment I did below, it is about changing the content of a file without having the need to do so. Could you please take a look at it. After it I will be happy to merge your PR!

File: graph/api/src/test/java/org/jboss/windup/graph/model/comparator/FilePathComparatorTest.java [hunk 1]
```diff
@@ -1,12 +1,13 @@
 package org.jboss.windup.graph.model.comparator;
 
+import org.jboss.windup.graph.model.comparator.FilePathComparator;
+import org.junit.Assert;
+import org.junit.Test;
```
[the remark this answers is not in the document]
Comment (jmle) [reply graph/api/src/test/java/org/jboss/windup/graph/model/comparator/FilePathComparatorTest.java]: @carlosthe19916 fixed

**Pull request merged**